app/Http/Controllers/V2/Dashboard/TotalTerrafundHeaderDashboardController.php

```diff
@@ -0,0 +1,143 @@
+            $query->where('uuid', $projectId);
+        }
+
+        return $query->count();
+    }
+
+    public function getTotalJobsCreatedSum(Request $request)
+    {
+        $projects = Project::where('framework_key', 'terrafund');
+        if ($request->has('country')) {
+            $country = $request->input('country');
+            $projects->where('country', $country);
+        } elseif ($request->has('uuid')) {
+            $projectId = $request->input('uuid');
+            $projects->where('uuid', $projectId);
+        }
+        $projects = $projects->pluck('id')->toArray();
+        $totalSum = 0;
+        foreach ($projects as $projectId) {
+            $reports = ProjectReport::where('project_id', $projectId)
+                ->sum(DB::raw('pt_total + ft_total'));
+            $totalSum += $reports;
+        }
+
+        return intval($totalSum);
+    }
+
+    public function getTotalHectaresRestoredGoalSum(Request $request)
+    {
+        $projects = Project::where('framework_key', 'terrafund');
+        if ($request->has('country')) {
+            $country = $request->input('country');
+            $projects->where('country', $country);
+        } elseif ($request->has('uuid')) {
+            $projectId = $request->input('uuid');
+            $projects->where('uuid', $projectId);
+        }
+
+        $total = $projects->sum('total_hectares_restored_goal');
+
+        return intval($total);
+    }
+
+    public function getTotalTreesRestoredSum(Request $request)
+    {
+        $projects = Project::where('framework_key', 'terrafund');
+
+        if ($request->has('country')) {
+            $country = $request->input('country');
+            $projects->where('country', $country);
+        } elseif ($request->has('uuid')) {
+            $projectId = $request->input('uuid');
+            $projects->where('uuid', $projectId);
+        }
+        $projects = $projects->get();
+
+        $totalSpeciesAmount = 0;
+
+        foreach ($projects as $project) {
+            $sites = $project->sites()->pluck('id')->toArray();
+            foreach ($sites as $siteId) {
+                $latestSiteReportId = SiteReport::where('site_id', $siteId)
+                    ->orderByDesc('due_at')
+                    ->value('id');
+                if ($latestSiteReportId !== null) {
+                    $totalSpeciesAmount += TreeSpecies::where('speciesable_id', $latestSiteReportId)->sum('amount');
+                }
+            }
+        }
+
+        return $totalSpeciesAmount;
+    }
+
+    public function getTotalTreesGrownGoalSum(Request $request)
+    {
+        $projects = Project::where('framework_key', 'terrafund');
+        if ($request->has('country')) {
+            $country = $request->input('country');
+            $projects->where('country', $country);
+        } elseif ($request->has('uuid')) {
+            $projectId = $request->input('uuid');
+            $projects->where('uuid', $projectId);
+        }
```

Comment:
This is a repeated in some of the methods, please can you add a helper function to simplify this?

For example a helper that would read all query params and then return the query builder.